misaligned.cpp

```diff
@@ -8,6 +14,9 @@ int printColorMap() {
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
             std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            number.push_back(i * 5 + j);
+            majorColorList.push_back(majorColor[i]);

```

Comment:
Nice way to make the behavior observable. Watch out with globals: The previous state is retained (which can influence multiple entry of the function, especially in multi-threaded situations)